perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -152,6 +154,48 @@ public AttributesManagerBlImpl(AttributesManagerImplApi attributesManagerImpl) {
 		return getAttributes(sess, resource, member, false);
 	}
 
+	@Override
+	public List<Attribute> getAttributes(PerunSession sess, Member member, Group group) throws InternalErrorException, WrongAttributeAssignmentException {
+		// get virtual attributes
+		List<Attribute> attributes = getAttributesManagerImpl().getVirtualAttributes(sess, member, group);
+		// filter out virtual attributes with null value
+		Iterator<Attribute> attributeIterator = attributes.iterator();
+		while(attributeIterator.hasNext()) if(attributeIterator.next().getValue() == null) attributeIterator.remove();
+		// adds non-empty non-virtual attributes
+		attributes.addAll(getAttributesManagerImpl().getAttributes(sess, member, group));
+		return attributes;
+	}
+
+	@Override
+	public List<Attribute> getAttributes(PerunSession sess, Member member, Group group, List<String> attrNames) throws InternalErrorException, WrongAttributeAssignmentException {
+		if(attrNames.isEmpty()) return new ArrayList<>();
+		// get virtual attributes
+		List<Attribute> attributes = getAttributesManagerImpl().getVirtualAttributes(sess, member, group);
```

Comment:
There is no filtering on virtual attributes. It always take all virtual attributes and filter only non-virtual attributes. Thats not correct behavior.